go.mod

```diff
@@ -71,13 +72,12 @@ require (
 	github.com/jbenet/go-temp-err-catcher v0.1.0
 	github.com/jbenet/goprocess v0.1.4
 	github.com/libp2p/go-doh-resolver v0.4.0
-	github.com/libp2p/go-libp2p v0.23.4
-	github.com/libp2p/go-libp2p-http v0.2.1
-	github.com/libp2p/go-libp2p-kad-dht v0.18.0
-	github.com/libp2p/go-libp2p-kbucket v0.4.7
-	github.com/libp2p/go-libp2p-loggables v0.1.0
+	github.com/libp2p/go-libp2p v0.24.0
+	github.com/libp2p/go-libp2p-http v0.4.0
+	github.com/libp2p/go-libp2p-kad-dht v0.18.1-0.20221120203245-43eacaff1f99
```

Comment:
@marten-seemann since this is still wip, is it possible to include https://github.com/libp2p/go-libp2p-kad-dht/pull/793?

Reply:
We released v0.19.0 yesterday.

https://github.com/libp2p/go-libp2p-kad-dht/pull/793 would require a new release, but this is orthogonal to this PR, isn't it?

Reply:
Yes, this could be a separate release and could be a separate PR.

I am flagging this because "required" list for Kubo 0.18 (https://github.com/ipfs/kubo/issues/9417)  includes https://github.com/ipfs/kubo/issues/9389 which (iiuc) needs a new libp2p release with https://github.com/libp2p/go-libp2p-kad-dht/pull/794 and https://github.com/libp2p/go-libp2p-kad-dht/pull/793. 

So either we need a patch release, or to descope the above from Kubo 0.18.   (cc @BigLep)

Reply:
> I am flagging this because "required" list for Kubo 0.18 (https://github.com/ipfs/kubo/issues/9417) includes https://github.com/ipfs/kubo/issues/9389 which (iiuc) needs a new libp2p release with https://github.com/libp2p/go-libp2p-kad-dht/pull/794 and https://github.com/libp2p/go-libp2p-kad-dht/pull/793.

go-libp2p-kad-dht is not a dependency of go-libp2p, it's the other way around. So no new go-libp2p release required.

Reply:
Ah, sweet!  In that case is it ok for me to just merge https://github.com/libp2p/go-libp2p-kad-dht/pull/794  and https://github.com/libp2p/go-libp2p-kad-dht/pull/793 and make a new v0.20.0 release of go-libp2p-kad-dht?

Reply:
@lidel I've bumped go-libp2p-kad-dht to v0.19.0 in https://github.com/ipfs/go-namesys/pull/33 as long as you don't break v0.19.0's api in v0.20.0 (which neither of the PR you linked do), I'm good.

Reply:
Pinged some folks for review on go-libp2p-kad-dht PRs, but it should not block this PR. 
We can bump in a follow-up one.

Reply:
@Jorropo before you merge this PR, please incorporate go-libp2p-kad-dht update from  https://github.com/ipfs/kubo/pull/9491  :pray: